app/adapters/whats_app_adapter/three_sixty_dialog_outbound.rb

```diff
@@ -114,17 +114,15 @@ def send_message_template(recipient, message)
       def send_message(recipient, message)
         files = message.files
 
-        if files.blank?
-          WhatsAppAdapter::ThreeSixtyDialogOutbound::Text.perform_later(organization_id: message.organization.id,
-                                                                        payload: text_payload(
-                                                                          recipient, message.text
-                                                                        ),
-                                                                        message_id: message.id)
-        else
-          files.each do |_file|
-            WhatsAppAdapter::ThreeSixtyDialog::UploadFileJob.perform_later(message_id: message.id)
-          end
-        end
+        WhatsAppAdapter::ThreeSixtyDialog::UploadFileJob.perform_later(message_id: message.id) if files.present?
+
+        return if message.text.blank?
+
+        WhatsAppAdapter::ThreeSixtyDialogOutbound::Text.perform_later(organization_id: message.organization.id,
+                                                                      payload: text_payload(
+                                                                        recipient, message.text
+                                                                      ),
+                                                                      message_id: message.id)
```

Comment:
I think it might be a better UX to upload the file, if one is attached, then send out the file and after it was sent successfully. Alternatively, I've since clarified that we can send the message text as a caption with a limit of 1024 characters. So, we could potentially do that. For telegram, we check if the message text is greater than the limit and if it is we send it separately.